src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -2165,14 +2175,26 @@ export class StudyViewPageStore
         // We do not support studyIds in the query filters
         let filters: Partial<StudyViewFilter> = {};
         if (query.filterJson) {
-            try {
-                filters = JSON.parse(
-                    decodeURIComponent(query.filterJson)
-                ) as Partial<StudyViewFilter>;
-                this.updateStoreByFilters(filters);
-            } catch (e) {
-                //  TODO: add some logging here?
+            const parsedFilterJson = this.parseRawFilterJson(query.filterJson);
```

Comment:
this is a comment about updateStoreFromURL and the "problem" precedes your work here, but I think we should try to address it.   the issue is that we are calling this async method without explicitly handling "asycnronicity".  I.e. we don't ever wait for it to finish.   It doesn't cause a problem because we await something which is dependent on this stuff.  But it's not super good and I think it would be pretty easy to get right.  Lets zoom for a few mins tomorrow and we can fix it.